### PR TITLE
Add JWT keys endpoint and use it in tests

### PR DESCRIPTION
**Pull request by achamayou**
Part of #2519, preferable to #2674 in my opinion.

### CODE REVIEW

File: src/node/rpc/member_frontend.h [hunk 1]
```diff
@@ -946,6 +946,24 @@ namespace ccf
         .set_openapi_hidden(true)
         .install();
 
+      using JWTKeyMap = std::map<JwtKeyId, Cert>;
+
+      auto get_jwt_keys = [this](auto& ctx, nlohmann::json&& body) {
```
Comment (jumaffre) [inline src/node/rpc/member_frontend.h:951]: We should also use this new endpoint in `jwt_issuer.py:wait_for_refresh`

File: src/node/rpc/member_frontend.h [hunk 2]
```diff
@@ -946,6 +946,24 @@ namespace ccf
...
+        JWTKeyMap kmap;
+        keys->foreach([&kmap](const auto& kid, const auto& kpem) {
+          kmap[kid] = kpem;
+          return false;
+        });
+
+        return make_success(kmap);
+      };
+      make_endpoint(
+        "jwt_keys/all", HTTP_GET, json_adapter(get_jwt_keys), no_auth_required)
```
Comment (jumaffre) [inline src/node/rpc/member_frontend.h:963]: Why not just `jwt_keys`?

File: src/node/rpc/member_frontend.h (hunk 1 above)
Comment (achamayou) [reply src/node/rpc/member_frontend.h:951]: I'd missed, but yes, absolutely.

File: src/node/rpc/member_frontend.h (hunk 2 above)
Comment (achamayou) [reply src/node/rpc/member_frontend.h:963]: No strong reason, other than there's precedent for /all elsewhere it seemed like it left more room for other GET endpoints at the same level (/issuers etc).

Happy to defer to a REST-qualified expert :)

File: src/node/rpc/member_frontend.h [hunk 3]
```diff
@@ -946,6 +946,24 @@ namespace ccf
...
         .install();
 
+      using JWTKeyMap = std::map<JwtKeyId, Cert>;
+
+      auto get_jwt_keys = [this](auto& ctx, nlohmann::json&& body) {
+        auto keys = ctx.tx.template ro<JwtPublicSigningKeys>(
+          ccf::Tables::JWT_PUBLIC_SIGNING_KEYS);
+        JWTKeyMap kmap;
+        keys->foreach([&kmap](const auto& kid, const auto& kpem) {
+          kmap[kid] = kpem;
```
Comment (jumaffre) [inline src/node/rpc/member_frontend.h:956]: Could we bundle the issuer here as well, by looking up the `public:ccf.gov.jwt.public_signing_key_issuer` table?

File: tests/jwt_test.py [hunk 4]
```diff
@@ -28,13 +28,20 @@ def test_refresh_jwt_issuer(network, args):
     return network
 
 
+def make_pem_cert(pem_encoded):
```
Comment (jumaffre) [inline tests/jwt_test.py:31]: Can't we just re-use the existing `infra.crypto.cert_der_to_pem` function?

Comment (achamayou) [reply tests/jwt_test.py:31]: It's not particularly convenient, we have to decode the string first. If anything I was more leaning towards making the endpoint do the wrapping so we return proper PEM certs.

File: src/node/rpc/member_frontend.h (hunk 3 above)
Comment (achamayou) [reply src/node/rpc/member_frontend.h:956]: We could, but we won't until there's a use case for it.

File: tests/jwt_test.py (hunk 4 above)
Comment (jumaffre) [reply tests/jwt_test.py]: Returning PEM directly sounds good!

Comment (letmaik) [reply tests/jwt_test.py]: +1 on PEM, as that allows to serve public key PEMs as well (if they should come in the future, see #2671) and keep a single response schema,

**Review by letmaik — APPROVED** [review]

File: tests/jwt_test.py (hunk 4 above)
Comment (achamayou) [reply tests/jwt_test.py]: Yes I've done that now.

**Review by jumaffre — APPROVED** [review]

**Pull request merged**